src/core/Akka.Cluster/VectorClock.cs

```diff
@@ -32,34 +31,74 @@ namespace Akka.Cluster
     /// </summary>
     public sealed class VectorClock
     {
-        /**
-         * Hash representation of a versioned node name.
-         */
-        //TODO: Node is alias for string in akka. Alternatives to below implementation?
-        //Do we really need to hash? Why not just use string and get rid of Node class		
+        private bool Equals(VectorClock other)
+        {
+            return IsSameAs(other);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            return obj is VectorClock && Equals((VectorClock) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 23;
+                foreach (var c in _versions)
+                {
+                    hashCode = hashCode * 31 + c.Key.GetHashCode();
+                    hashCode = hashCode * 31 + c.Value.GetHashCode();
+                }
+                return hashCode;
+            }
+        }
+
         /// <summary>
-        /// TBD
+        /// Hash representation of a versioned node name.
         /// </summary>
         public class Node : IComparable<Node>
         {
-            readonly string _value;
+            private readonly string _value;
+            private readonly int _computedHashValue;
 #if DEBUG
             string ActualValue { get; set; }
 #endif
             /// <summary>
-            /// TBD
+            /// Creates a new vectorclock node.
             /// </summary>
-            /// <param name="value">TBD</param>
+            /// <param name="value">The hash value of the string.</param>
             public Node(string value)
             {
                 _value = value;
+                _computedHashValue = 23;
+                /*
+                 * IMPORTANT: in a distributed system, relying on the underlying object.GetHashCode function
+                 * may return inconsistent values across servers. Therefore we want to compute our hashcode from 
+                 * the sum of the parts and use that instead, as that hashcode value is determined solely by the
+                 * content of the string and doesn't include any other environmental factors.
+                 * 
+                 * NOTE: Aaronontheweb, 4/7/2017 ideally this information should be computed before we create the VectorClock
+                 * at the same time as the hash is taken, but I decided to implement it here so we wouldn't have to change the
+                 * constructor signature, which would affect wire format compatibility.
+                 */
+                unchecked
+                {
+                    foreach (var c in value)
+                    {
+                        _computedHashValue *= _computedHashValue * 31 + c; // using the byte value of each char
```

Comment:
This may no longer be necessary.... was an early theory I had as to why the vector clocks didn't sort properly. Turned out to be wrong.